brion/circuitConfig.cpp

```diff
@@ -64,7 +64,14 @@ struct CircuitConfig::Impl
         : resolver(uri)
     {
         const auto json = parseSonataJson(uri);
-        targetSimulator = json["target_simulator"];
+        try
+        {
+            targetSimulator = json.at("target_simulator");
+        }
+        catch (nlohmann::detail::exception&)
```

Comment:
I think you should catch a more specific exception. I think the out_of_range is the right one since you only want to catch the missing key exception.

Reply:
Done